clang/lib/Driver/ToolChains/SYCL.cpp

```diff
@@ -106,16 +118,32 @@ const char *SYCL::Linker::constructLLVMLinkCommand(
   // an actual object/archive.  Take that list and pass those to the linker
   // instead of the original object.
   if (JA.isDeviceOffloading(Action::OFK_SYCL)) {
-    auto SYCLDeviceLibIter =
-        std::find_if(InputFiles.begin(), InputFiles.end(), [](const auto &II) {
-          StringRef InputFilename =
-              llvm::sys::path::filename(StringRef(II.getFilename()));
-          if (InputFilename.startswith("libsycl-") &&
-              InputFilename.endswith(".o"))
-            return true;
-          return false;
-        });
-    bool LinkSYCLDeviceLibs = (SYCLDeviceLibIter != InputFiles.end());
+    auto isSYCLDeviceLib = [&C](const InputInfo &II) {
+      const ToolChain *HostTC = C.getSingleOffloadToolChain<Action::OFK_Host>();
+      StringRef LibPostfix = ".o";
+      if (HostTC->getTriple().isWindowsMSVCEnvironment() &&
+          C.getDriver().IsCLMode())
```

Comment:
I may have swayed you the wrong way in email, we should only need the triple check.  CL Mode is specific to `clang-cl`
```suggestion
      if (HostTC->getTriple().isWindowsMSVCEnvironment())
```

Reply:
Hi, @mdtoguchi 
It seems checking MSVC triple only won't meet our requirement. Users can use "clang" driver on Windows platform and compiler driver will not be in CL mode(instead, it will be in CXX or CC mode). If clang driver is used on Windows, compiler will use ".o" as postfix instead of ".obj".
Only when "clang-cl" is used on Windows, the ".obj" postfix will be used.
Thanks very much.